### PR TITLE
Np 70 fe booking detail

### PR DESCRIPTION
**Pull request by maxkanin**
## **Ticket**

**JIRA ID:** [NP-70](https://parking-system.atlassian.net/browse/NP-70)

## Changes proposed in this pull request
- [IMPORTANT] Reviewer **will not review** any PR without summary of changes
- add booking detail ui
- connect booking summary with booking detail

please fill all the required fields from below Checklist otherwise reviewer **will not review** this PR
## Checklist
- [ ] Screenshot evidence for both IOS and Android
- [ ] Total change is less than 500 lines of code

## **Screenshot Evidence**

- ios

https://github.com/ParkingSystemWithLPR/Napark-application/assets/88888222/579c659c-10d1-401f-a839-6efad40d473c

- android

https://github.com/ParkingSystemWithLPR/Napark-application/assets/88888222/ae4b06d7-7a9a-4034-9300-02be03de522c

[NP-70]: https://parking-system.atlassian.net/browse/NP-70?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ingthanhunt** [comment]
ยังไม่ได้ดู code นะแต่อยากให้แก้ style นิดหน่อย คือ ios มันดูแน่นไปหน่อยอยากให้ลด fontSize แล้วก็อาจจะเพิ่ม gap ขึ้นกว่านี้ครับ

**Comment by maxkanin** [comment]
> ยังไม่ได้ดู code นะแต่อยากให้แก้ style นิดหน่อย คือ ios มันดูแน่นไปหน่อยอยากให้ลด fontSize แล้วก็อาจจะเพิ่ม gap ขึ้นกว่านี้ครับ

เดี๋ยวทำครับ

File: components/booking/BookingCardSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+const BookingCardSummary: React.FC<BookingCardSummaryProps> = ({
+  checkInDate,
+  checkInTime,
+  specification,
+  openSetting,
+}) => {
+  type DetailProps = {
+    topic: string;
+    value: string;
+  };
```
Comment (TaeCV) [inline components/booking/BookingCardSummary.tsx]: Move out of the component

File: components/booking/BookingCardSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const Detail: React.FC<DetailProps> = ({ topic, value }) => {
+    return (
+      <View style={styles.detailContainer}>
+        <BodyText text={topic} />
+        <BodyText text={value} />
+      </View>
+    );
+  };
```
Comment (TaeCV) [inline components/booking/BookingCardSummary.tsx]: - Wrap it with useCallback
- I prefer this one to be a function

File: components/booking/BookingCardSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            text={"Booking Details"}
+            containerStyle={styles.headerStyle}
+          />
+        </View>
+        <Pressable onPress={openSetting}>
+          <Ionicons name="settings-outline" style={styles.icon} />
+        </Pressable>
+      </View>
+      <View style={styles.propertyContainer}>
+        <Detail topic={"Check-in:"} value={checkInDate + " " + checkInTime} />
```
Comment (TaeCV) [inline components/booking/BookingCardSummary.tsx]: Make it as utils

File: components/booking/BookingDetailComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+          isRequired={true}
+        />
+        <TimeInput
+          title={""}
+          value={checkOutTime}
+          onTimeChange={checkOutTimeHandler}
+          outerContainerStyle={styles.timeContainer}
+          editable={isCheckOutTimeEditable}
+        />
+      </View>
```
Comment (TaeCV) [inline components/booking/BookingDetailComponent.tsx:153]: Add gap here

File: components/booking/BookingDetailComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        items={[
+          { label: "กข1234", value: "กข1234" },
+          {
+            label: "Not found your license plate",
+            value: "Not found your license plate",
+          },
+        ]}
+        title="License Plate"
+        isRequired={true}
+      ></DropdownInput>
```
Comment (TaeCV) [inline components/booking/BookingDetailComponent.tsx]: You can use <X />

File: screens/booking/BookingDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+            <BookingDetailComponent
+              licensePlate={bookingRequest.licensePlate}
+              setLicensePlate={(value: string) => {
+                if (value == "Not found your license plate") {
+                  handleOnChange("licensePlate", "");
+                  navigation.navigate("OtherStack", {
+                    screen: "CarInfoSetup",
+                    params: {
+                      mode: ActionMode.CREATE,
+                    },
```
Comment (TaeCV) [inline screens/booking/BookingDetail.tsx]: I think you should handle some kind of these stuffs in its component. Note: You can use `useNavigation()` to have a navigation there.

**Comment by maxkanin** [comment]
ios
https://github.com/ParkingSystemWithLPR/Napark-application/assets/88888222/98159639-2b1f-4b57-9e07-08f2a90db660
android
https://github.com/ParkingSystemWithLPR/Napark-application/assets/88888222/b0e6202c-d64e-441e-a392-1f589ac95bb2

File: components/booking/Specification.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    borderColor: Colors.red[400],
+    paddingVertical: 10,
+  },
+  headerContainer: {
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  parkingOutline: { borderRadius: 20, borderWidth: 1, alignItems: "center" },
+  header: { marginLeft: 5, alignSelf: "flex-start" },
+  radioButton: { alignItems: "flex-start", marginVertical: 10 },
```
Comment (TaeCV) [inline components/booking/Specification.tsx]: Do we really need flex-start here?

File: components/booking/Specification.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+              containerStyle={styles.header}
+            />
+          </View>
+        </View>
+        <RadioGroup
+          radioButtons={radioButtons}
+          onPress={onChange}
+          selectedId={specification}
+          containerStyle={styles.radioButton}
+        />
```
Comment (TaeCV) [inline components/booking/Specification.tsx:61]: Can you change the radio button to be Colors.red[400]?

**Comment by maxkanin** [comment]
android
https://github.com/ParkingSystemWithLPR/Napark-application/assets/88888222/eb0f5a90-2308-4015-926f-db5e7750d708
ios
https://github.com/ParkingSystemWithLPR/Napark-application/assets/88888222/e7c6a53f-2466-4cf4-9447-5f1077bde4e7

**Comment by TaeCV** [comment]
LGTM

**Pull request merged**